src/main/java/org/jitsi/srtp/crypto/Aes.java

```diff
@@ -872,27 +894,46 @@ public static synchronized Provider getProvider()
 
                     if (Provider.class.isAssignableFrom(clazz))
                     {
-                        Constructor<?> contructor
-                            = clazz.getConstructor(String.class);
-
-                        // The SunPKCS11 Config name should be unique in order
-                        // to avoid repeated initialization exceptions.
-                        String name = null;
-                        Package pkg = Aes.class.getPackage();
-
-                        if (pkg != null)
-                            name = pkg.getName();
-                        if (name == null || name.length() == 0)
-                            name = "org.jitsi.srtp";
-
-                        provider
-                            = (Provider)
-                                contructor.newInstance(
-                                        "--name=" + name + "\\n"
-                                            + "nssDbMode=noDb\\n"
-                                            + "attributes=compatibility");
+                        try
+                        {
+                            /* Java 9+.  PKCS11 configuration is set using the
+                             * configure method.
+                             * Use reflection so we can build with JDK 8. */
+                            Method configureMethod =
```

Comment:
Why not move this inside the null-check?

Reply:
Because I don't need/want to call `Provider.getPrototype()` if `getMethod("configure")` throws, as it will in Java 1.8.